### DISCUSSION TITLE
Using a target size (torch.Size([80670, 1])) that is different to the input size (torch.Size([80670]))

### DISCUSSION THREAD

**Discussion by khanwa** [pyg-team/pytorch_geometric#6046] [Q&A]
I want to implement GCN with my own dataset (from Pytorch geometric tutorial). I copied my notebook below. It is working when I use the original implementation, However, if I want to use custom data then it gives the below error:

[My notebook ](https://colab.research.google.com/drive/1-UyUTsZ4YEwfy1qLmqU6JKFbySYVF_OD?usp=sharing)

[Original implementation](https://colab.research.google.com/drive/14OvFnAXggxB8vM4e8vSURUp1TaKnovzX?usp=sharing)

I used Load[ CSV from pyG](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/load_csv.py) and tried to incorporate it into their GCN implementation. I know in the original implementation they used the same dataset but I want to try with custom data (load datasets for later).

Using a target size (torch.Size([80670, 1])) that is different to the input size (torch.Size([80670])). This will likely lead to incorrect results due to broadcasting. Please ensure they have the same size.

The problem arises: Print (data)

```
HeteroData(
  movie={ x=[9742, 404] },
  user={ num_nodes=610 },
  (user, rates, movie)={
    edge_index=[2, 100836],
    edge_label=[100836] # mine is edge_label=[100836,1]
  }
)
```
I don't know how to make edge_label=[100836,1] to edge_label=[100836] which will hopefully solve the issue.

Thank you.

**Comment by rusty1s** [comment]
You can just run `edge_label_index.view(-1)` to flatten your tensor.

**Reply by khanwa** [reply]
Thank you very much for the answer. What I tried is to use the load_csv.py as the data loader and used the (GCN) from bipartite_sage.py. But it seems to have other issues as well:
Now I got an error in  x_dict['user'] = self.user_emb(x_dict['user']) as **KeyError: 'user'** from    
 def forward(self, x_dict, edge_index_dict, edge_label_index).
Here is the [Colab Link](https://colab.research.google.com/drive/1-UyUTsZ4YEwfy1qLmqU6JKFbySYVF_OD?usp=sharing)

Thank you again.

**Reply by rusty1s** [reply]
It needs to be
```python
data['user'].x = torch.arange(user_mapping)
```
since you want to use the node IDs of users as input to the `Embedding` layer in the model.
